engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1130,6 +1138,136 @@ namespace dmGameSystem
+     *      physics.set_group("#collisionobject", "enemy")
+     * end
+     * ```
+     */
+    static int Physics_SetGroup(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        
+        if (! dmGameSystem::SetCollisionGroup(comp_world, comp, group_id)) {
+            return luaL_error(L, "Collision group not registered: %s.", dmHashReverseSafe64(group_id));
+        }
+
+        return 0;
+    }    
+
+    /*# returns the group of a collision object
+     *
+     * Returns the group name of a collision object as a hash.
+     *
+     * @name physics.get_group
+     * @param url [type:string|hash|url] the collision object to return the group of.
+     * @return [type:hash] hash value of the group.
+     * ```lua
+     * function checkIsEnemy()
+     *     local grp = physics.get_group("#collisionobject")
+     *     assert( grp == hash("enemy") )
+     * end
+     * ```
+     */    
+    static int Physics_GetGroup(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        
+        dmhash_t group_hash = dmGameSystem::GetCollisionGroup(comp_world, comp);
+        dmScript::PushHash(L, group_hash);
+        
+        return 1;
+    }    
+
+    /*# updates the mask of a collision object
+     *
+     * Sets or clears the masking of a group (maskbit) in a collision object.
+     *
+     * @name physics.set_maskbit
+     * @param url [type:string|hash|url] the collision object to change the mask of.
+     * @param group [type:string] the name of the group (maskbit) to modify in the mask.
+     * @param [type:boolean] boolean value of the new maskbit. 'true' to enable, 'false' to disable.
+     * ```lua
+     * function makeUserAlly()
+     *     -- no longer collide with the "user" group
+     *     physics.set_maskbit("#collisionobject","user",false)
+     * end
+     * ```
+     */
+    static int Physics_SetMaskBit(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        bool boolvalue = CheckBoolean(L, 3);
+        
+        if (! dmGameSystem::SetCollisionMaskBit(comp_world, comp, group_id, boolvalue)) {
+            return luaL_error(L, "Collision group not registered: %s.", dmHashReverseSafe64(group_id));
```

Comment:
The `DM_LUA_STACK_CHECK` should be used in tandem with `DM_LUA_ERROR`.
Otherwise the stackchecker won't do its job of making sure that there are noLUa objects pushed onto the stack.

I notice that there already is a mix of DM_LUA_STACK_CHECK/luaL_error in the scirpt_physics, and that's not right either sadly.